jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/CombinedResource.java

```diff
@@ -127,127 +153,67 @@ public Resource resolve(String subUriPath)
-        Resource addedResource = null;
+        Resource resolved = null;
         for (Resource res : _resources)
         {
-            addedResource = res.resolve(subUriPath);
-            if (Resources.missing(addedResource))
+            resolved = res.resolve(subUriPath);
+            if (Resources.missing(resolved))
                 continue; // skip, doesn't exist
-            if (!addedResource.isDirectory())
-                return addedResource; // Return simple (non-directory) Resource
+            if (!resolved.isDirectory())
+                return resolved; // Return simple (non-directory) Resource
             if (resources == null)
                 resources = new ArrayList<>();
-            resources.add(addedResource);
+            resources.add(resolved);
         }
 
         if (resources == null)
-            return addedResource; // This will not exist
+            return resolved; // This will not exist
 
         if (resources.size() == 1)
             return resources.get(0);
 
-        return new ResourceCollection(resources);
+        return new CombinedResource(resources);
     }
 
     @Override
     public boolean exists()
     {
-        for (Resource r : _resources)
-        {
-            if (r.exists())
-            {
-                return true;
-            }
-        }
-
-        return false;
+        return _resources.stream().anyMatch(Resource::exists);
     }
 
     @Override
     public Path getPath()
     {
-        for (Resource r : _resources)
-        {
-            Path p = r.getPath();
-            if (p != null)
-                return p;
-        }
-        return null;
-    }
-
-    @Override
-    public InputStream newInputStream() throws IOException
-    {
-        for (Resource r : _resources)
-        {
-            if (!r.exists())
-            {
-                // Skip, cannot open anyway
-                continue;
-            }
-            InputStream is = r.newInputStream();
-            if (is != null)
-            {
-                return is;
-            }
-        }
-
-        throw new FileNotFoundException("Resource does not exist");
-    }
-
-    @Override
-    public ReadableByteChannel newReadableByteChannel() throws IOException
-    {
-        for (Resource r : _resources)
-        {
-            ReadableByteChannel channel = r.newReadableByteChannel();
-            if (channel != null)
-            {
-                return channel;
-            }
-        }
         return null;
     }
 
     @Override
     public String getName()
     {
-        for (Resource r : _resources)
-        {
-            String name = r.getName();
-            if (name != null)
-            {
-                return name;
-            }
-        }
         return null;
     }
 
     @Override
     public String getFileName()
     {
+        String filename = null;
+        // return a non-null filename only if all resources agree on the same name.
```

Comment:
Perhaps we should make `getFileName()` not be the last segment name in case of Directory?   Is there a use case were in we care about, and use `getFileName()` in a directory case?

Reply:
We can use the fileName to generate a context name.  I'm not 100% sure we still use this method for that, but it is a reasonable use-case to ask a directory for the name of it's last segment.

This method exists and I think this is a reasonable interpretation of it for a combined resource.